lib/appsignal/phoenix/instrumenter.ex

```diff
@@ -52,34 +26,50 @@ if Appsignal.phoenix? do
 
     require Logger
 
-    import Appsignal.Phoenix.InstrumenterDSL
-    instrumenter :phoenix_controller_call
-    instrumenter :phoenix_controller_render
-
     @doc false
-    def maybe_transaction_start_event(%Transaction{} = transaction, args) do
+    def phoenix_controller_call(:start, _compiled, args) do
+      id = Logger.metadata()[:request_id] || Transaction.generate_id()
+      transaction = Transaction.start(id, :http_request)
+
       {Transaction.start_event(transaction), args}
     end
-    def maybe_transaction_start_event(pid, args) when is_pid(pid) do
-      maybe_transaction_start_event(Appsignal.TransactionRegistry.lookup(pid), args)
-    end
-    def maybe_transaction_start_event(%{conn: %Plug.Conn{} = conn}, args) do
-      maybe_transaction_start_event(conn.assigns[:appsignal_transaction], args)
-    end
-    def maybe_transaction_start_event(%{}, args) do
-      maybe_transaction_start_event(self(), args)
+
+    @doc false
+    def phoenix_controller_call(:stop, _diff, {%Transaction{} = transaction, %{conn: conn} = args}) do
+      Transaction.finish_event(
+        transaction,
+        "phoenix_controller_call",
+        "phoenix_controller_call",
+        args,
```

Comment:
On master, I explicitly remove `:conn` from `args` before sending it to `finish_event` (see the `cleanup_args` function)